### PR TITLE
Nexus 4635 nexus version

### PR DESCRIPTION
**Pull request by velo**
Add nexusVersion to nexus.xml

### CODE REVIEW

File: nexus/nexus-api/src/main/java/org/sonatype/nexus/proxy/events/NexusVersionChangedEvent.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+package org.sonatype.nexus.proxy.events;
+
+/**
+ * This event is fired every first time a new nexus version is launched
+ * 
+ * @author velo
+ */
+public class NexusVersionChangedEvent
+    extends AbstractVetoableEvent<Object>
```
Comment (cstamas) [inline nexus/nexus-api/src/main/java/org/sonatype/nexus/proxy/events/NexusVersionChangedEvent.java]: a) bad named event: NexusUpgradedEvent would be better unless we plan to programmatically support "downgrading" too?
b) why is this Vetoable event? Who can put veto on this and why?

File: nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/ApplicationConfigurationSource.java [hunk 2]
```diff
@@ -48,4 +48,14 @@ public interface ApplicationConfigurationSource
     void backupConfiguration()
         throws IOException;
 
+    /**
+     * Is nexus instance upgraded
+     */
+    boolean isInstanceUpgraded();
```
Comment (cstamas) [inline nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/ApplicationConfigurationSource.java:54]: This should go to ConfigurationSource, but it's kept as hostage (by security) in base-config :(

So, for now should be okay.

File: nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/ApplicationConfigurationSource.java [hunk 3]
```diff
@@ -48,4 +48,14 @@ public interface ApplicationConfigurationSource
...
+
+    /**
+     * Setter for nexus instance upgraded
+     */
+    void setInstanceUpgraded( boolean instanceUpgraded );
```
Comment (cstamas) [inline nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/ApplicationConfigurationSource.java]: Is setter really needed?

File: nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/FileConfigurationSource.java [hunk 4]
```diff
@@ -201,6 +210,26 @@ public class FileConfigurationSource
...
+        if ( currentVersion.equals( previousVersion ) )
+        {
+            setInstanceUpgraded( false );
+        }
+        else
+        {
+            setInstanceUpgraded( true );
+            getConfiguration().setNexusVersion( currentVersion );
+            storeConfiguration();
+            eventMulticaster.notifyEventListeners( new NexusVersionChangedEvent( this, previousVersion,
```
Comment (cstamas) [inline nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/FileConfigurationSource.java]: Not sure i'd go with events (yet). Model upgrade was not emitting events either, and Status flags with this change would be properly set.

Let's go with baby steps, and not introduce events for these yet (unless some other good reason mentioned for it's need)

Comment (velo) [inline nexus/nexus-configuration/src/main/java/org/sonatype/nexus/configuration/source/FileConfigurationSource.java]: Ok, will nuke the event

**Comment by cstamas** [comment]
Even better with these changes :)

**Comment by bdemers** [comment]
+1

**Pull request merged**